### PR TITLE
Remove ordering traits from `HirId`

### PR DESCRIPTION
**Pull request by pierwill**
See https://github.com/rust-lang/rust/issues/90317.

### CODE REVIEW

**Comment by pierwill** [comment]
First things first: remove all the sorting and see what breaks. 😈

File: compiler/rustc_mir_transform/src/inline.rs [hunk 1]
```diff
@@ -206,17 +202,7 @@ impl<'tcx> Inliner<'tcx> {
...
-            // Avoid a cycle here by only using `instance_mir` only if we have
-            // a lower `HirId` than the callee. This ensures that the callee will
-            // not inline us. This trick only works without incremental compilation.
-            // So don't do it if that is enabled.
-            if !self.tcx.dep_graph.is_fully_enabled() && self.hir_id.index() < callee_hir_id.index()
-            {
-                return Ok(());
-            }
-
+        if callee.def_id().as_local().is_some() {
```
Comment (cjgillot) [inline compiler/rustc_mir_transform/src/inline.rs:205]: Marking as blocked on https://github.com/rust-lang/rust/pull/85321.

File: compiler/rustc_hir/src/hir_id.rs [hunk 2]
```diff
@@ -44,18 +40,6 @@ impl fmt::Display for HirId {
...
-        (self.index()).cmp(&(other.index()))
-    }
-}
-
-impl PartialOrd for HirId {
-    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        Some(self.cmp(&other))
-    }
-}
-
```
Comment (BGR360) [inline compiler/rustc_hir/src/hir_id.rs:58]: Can you add a comment to `HirId` for posterity, so nobody tries adding these traits in the future?

Comment (pierwill) [reply compiler/rustc_hir/src/hir_id.rs:58]: Great idea, @BGR360! Thanks.

Comment (Aaron1011) [reply compiler/rustc_hir/src/hir_id.rs:58]: You can also add a negative impl (`impl !PartialOrd for HirId`) which will cause an error if someone tries to add back a regular impl.

Comment (pierwill) [reply compiler/rustc_hir/src/hir_id.rs:58]: Amazing! Thanks @Aaron1011 😇

File: compiler/rustc_mir_transform/src/inline.rs (hunk 1 above)
Comment (bjorn3) [reply compiler/rustc_mir_transform/src/inline.rs:213]: https://github.com/rust-lang/rust/pull/85321 is now merged. Removing S-blocked.

**Comment by Dylan-DPC** [comment]
@pierwill can you address the CI failure? thanks

**Comment by Dylan-DPC** [comment]
Closing this as inactive. Feel free to reöpen or preferably create a new pull request

**Pull request closed without merging**